src/lib.rs

```diff
@@ -1921,6 +1921,16 @@ impl Connection {
 
         self.max_rx_data_next = self.max_rx_data_next.saturating_add(read as u64);
 
+        let readable = stream.readable();
```

Comment:
It seems redundant to assign`readable` here; the code suggestion below just calls the function directly.

However, after looking through the whole PR, I wonder if we really need to expose these `push_readable()` and `clear_readable()` methods at all. We could just expose a single function like `set_readable(id: u64, readable: bool)`, which would reduce code size and avoid leaking stream.rs implementation detail.

Reply:
The assignment is yet again because of our friend the borrow checker. Without the assignment the `stream` variable would keep the borrow on `self` "alive", so we wouldn't be able to call `self.streams.mark_readable()` and the like.

Anyway, I merged the various `push_*()` and `clear_*()` into single `mark_*()` methods.